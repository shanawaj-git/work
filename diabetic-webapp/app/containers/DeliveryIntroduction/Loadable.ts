/**
 *  Asynchronoulsy load the component from Preview Prescription Screen
 */

import loadable from '../../utils/loadable';
export default loadable(() => import('./index'));
