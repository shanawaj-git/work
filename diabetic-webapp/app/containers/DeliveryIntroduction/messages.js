import { defineMessages } from 'react-intl';
export default defineMessages({
  deliveryNoteHeaderLabel: {
    id: 'delivery.header.label',
    defaultMessage: 'Be present at',
  },
  deliveryNoteHeaderSuffixLabel: {
    id: 'delivery.header.label.suffix',
    defaultMessage: 'the time of delivery ',
  },
  deliveryNoteDescription: {
    id: 'delivery.description.label',
    defaultMessage:
      '<p>You are required to <strong>present in person </strong>at the time of delivery.Please select a date and the time accordingly.</p>',
  },
  deliveryStep1Title: {
    id: 'delivery.step1.title.label',
    defaultMessage: 'Step 1',
  },
  deliveryStep1TitleDescription: {
    id: 'delivery.step1.title.label.description',
    defaultMessage:
      'Select a date,time and place for delivery,so you can receive your meds in person.',
  },
  deliveryStep2Title: {
    id: 'delivery.step1.title.label',
    defaultMessage: 'Step 2',
  },
  deliveryStep2TitleDescription: {
    id: 'delivery.step2.title.label.description',
    defaultMessage:
      'Setup your refill reminder so your health doesnt miss out ',
  },
  deliveryIUnderstandButton: {
    id: 'delivery.i.understand.button',
    defaultMessage: 'I Understand,Proceed',
  },
  deliveryWhyIBePresentButton: {
    id: 'delivery.i.be.present.button',
    defaultMessage: 'Why must i be present in person?',
  },
});
