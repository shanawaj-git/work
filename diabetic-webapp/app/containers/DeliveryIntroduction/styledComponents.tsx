import styled from 'styled-components';
import { Typography } from '@albathanext/design-system';

export const DeliveryNoteScreen = styled.div`
  display: flex;
  flex-direction: column;
  align-items: start;
  height: 100%;
  width: 100%;
  justify-content: space-around;
  padding: 0px 18px;
  align-items: center;
`;
export const DeliveryNoteDescription = styled.div`
  align-self: start;
`;

export const DeliveryNoteHeader = styled.div`
  display: inline-block;
  clear: both;
  overflow: hidden;
  white-space: nowrap;
  align-self: start;
`;
export const ButtonContainer = styled.div`
  width: 100%;
  .MuiButton-root {
    border-radius: 12px;
    width: 100%;
  }
  font-family: 'Neue Haas Grotesk Display Pro';
  font-style: normal;
  font-weight: 600;
  font-size: 10px;
  line-height: 20px;
  text-align: center;
  letter-spacing: 0.4px;
`;

export const ClickHereLink = styled(Typography)`
  font-family: 'Neue Haas Grotesk Display Pro';
  font-style: normal;
  font-weight: 500;
  font-size: 15px;
  line-height: 22px;
  text-align: center;
  letter-spacing: 0.3px;
  color: #12a1c0;
  width: 100%;
  padding-bottom: 5%;
`;

export const DeliveryStepsContainer = styled.div`
  display: flex;
  flex-direction: row;
  align-items: center;
`;

export const DeliveryStepImageContainer = styled.div`
  display: flex;
  flex: 1;
  flex-wrap: wrap;
  align-items: center;
  justify-content: center;
  width: 100px;
  height: 100px;
  background: #ebebe8;
  border: 1px solid #d4d9d9;
  border-radius: 12px;
`;
export const DeliveryStepTextContainer = styled.div`
  display: flex;
  flex: 3;
  flex-direction: column;
  justify-content: space-between;
  margin-left: 16px;
  margin-right: 16px;
`;
export const StepTitle = styled(Typography)`
  font-family: 'Neue Haas Grotesk Display Pro';
  font-weight: 600;
  font-size: 18px;
  line-height: 24px;
  letter-spacing: 0.6px;
  color: #00664f;
  font-weight: bold;
`;
export const StepDescription = styled(Typography)`
  font-family: 'Neue Haas Grotesk Display Pro';
  font-weight: 500;
  font-size: 13px;
  line-height: 22px;
  letter-spacing: 0.3px;
  color: #495151;
  align-self: stretch;
  width: 211px;
`;
export const Margin = styled.div`
  width: 80%;
  border: 1px solid #d4d9d9;
  align-items: center;
`;

export const DeliveryDescription = styled(Typography)`
  width: 327px;
  height: 66px;
  font-family: 'Neue Haas Grotesk Display Pro';
  font-style: normal;
  font-weight: 500;
  font-size: 15px;
  line-height: 22px;
  letter-spacing: 0.3px;
  color: #495151;
`;

export const DeliveryHeaderLabel = styled(Typography)`
  font-family: 'Neue Haas Grotesk Display Pro';
  font-style: normal;
  font-size: 32px;
  line-height: 36px;
  font-weight: 600;
  letter-spacing: 0.6px;
  line-height: 36px;
  color: #020a0a;
`;
export const DeliveryHeaderSuffixLabel = styled(Typography)`
  font-family: 'Neue Haas Grotesk Display Pro';
  font-style: normal;
  font-size: 36px;
  line-height: 36px;
  font-weight: 500;
  letter-spacing: 0.6px;
  line-height: 36px;
  color: #020a0a;
`;
