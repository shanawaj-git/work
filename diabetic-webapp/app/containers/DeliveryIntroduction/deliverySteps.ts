import messages from './messages';
const deliveryStepImage1 = require('images/deliveryStepImage1.png');
const deliveryStepImage2 = require('images/deliveryStepImage2.png');

export const deliverySteps = [
  {
    id: 1,
    deliveryStepTitle: messages.deliveryStep1Title,
    deliveryStepTitleDescription: messages.deliveryStep1TitleDescription,
    deliveryImage: deliveryStepImage1,
  },
  {
    id: 2,
    deliveryStepTitle: messages.deliveryStep2Title,
    deliveryStepTitleDescription: messages.deliveryStep2TitleDescription,
    deliveryImage: deliveryStepImage2,
  },
];
