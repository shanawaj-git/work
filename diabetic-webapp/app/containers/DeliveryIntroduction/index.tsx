import React, { useState } from 'react';
import PropTypes, { any, InferProps } from 'prop-types';
import { injectIntl } from 'react-intl';
import { TypographyType, Button } from '@albathanext/design-system';
import ReactHtmlParser from 'react-html-parser';
import messages from './messages';
import { deliverySteps } from './deliverySteps';
import DeliveryModal from '../../components/organism/PresentInPersonReasonModal';
import {
  DeliveryNoteScreen,
  DeliveryNoteDescription,
  DeliveryNoteHeader,
  ButtonContainer,
  ClickHereLink,
  DeliveryStepsContainer,
  DeliveryStepImageContainer,
  DeliveryStepTextContainer,
  StepTitle,
  StepDescription,
  Margin,
  DeliveryHeaderLabel,
  DeliveryDescription,
  DeliveryHeaderSuffixLabel,
} from './styledComponents';
const ADDRESS_SELECTION_SCREEN = 'address-selection';
function DeliveryIntroduction({ intl, history }: DeliveryNoteTypes) {
  const [openInPersonReasonModal, setOpen] = useState(false);

  const openModel = () => setOpen(true);
  const closeModel = () => setOpen(false);

  const deliveryAddressSelection = async () => {
    history.push({
      pathname: ADDRESS_SELECTION_SCREEN,
    });
  };

  const DeliveryStep = ({ step }: { step: any }) => (
    <DeliveryStepsContainer>
      <DeliveryStepImageContainer>
        <img src={step.deliveryImage} alt="header_img" />
      </DeliveryStepImageContainer>
      <DeliveryStepTextContainer>
        <StepTitle typographyType={TypographyType.TITLE_4}>
          {intl.formatMessage(step.deliveryStepTitle)}
        </StepTitle>
        <StepDescription typographyType={TypographyType.SUB_HEAD}>
          {intl.formatMessage(step.deliveryStepTitleDescription)}
        </StepDescription>
      </DeliveryStepTextContainer>
    </DeliveryStepsContainer>
  );

  return (
    <DeliveryNoteScreen>
      <DeliveryNoteHeader>
        <DeliveryHeaderLabel typographyType={TypographyType.TITLE_1}>
          {intl.formatMessage(messages.deliveryNoteHeaderLabel)}
        </DeliveryHeaderLabel>
        <DeliveryHeaderSuffixLabel typographyType={TypographyType.LARGE_TITLE}>
          {intl.formatMessage(messages.deliveryNoteHeaderSuffixLabel)}
        </DeliveryHeaderSuffixLabel>
      </DeliveryNoteHeader>
      <DeliveryNoteDescription>
        <DeliveryDescription
          typographyType={TypographyType.SUB_HEAD}
          component="span"
        >
          {ReactHtmlParser(
            intl.formatMessage(messages.deliveryNoteDescription),
          )}
        </DeliveryDescription>
      </DeliveryNoteDescription>
      <Margin />
      {deliverySteps.map(step => (
        <DeliveryStep step={step} key={step.id} />
      ))}
      <ButtonContainer>
        <Button
          data-testid="iUnderstand-btn"
          variant="contained"
          size="large"
          onClick={deliveryAddressSelection}
        >
          <>{intl.formatMessage(messages.deliveryIUnderstandButton)}</>
        </Button>
      </ButtonContainer>
      <ClickHereLink>
        <button
          type="button"
          id="contactSupport"
          onClick={openModel}
          data-testid="modal-link"
        >
          {intl.formatMessage(messages.deliveryWhyIBePresentButton)}
        </button>
      </ClickHereLink>
      <DeliveryModal open={openInPersonReasonModal} closeHandler={closeModel} />
    </DeliveryNoteScreen>
  );
}

const DeliveryNotePropTypes = {
  intl: any,
  history: PropTypes.shape({
    push: PropTypes.func.isRequired,
  }),
};

type DeliveryNoteTypes = InferProps<typeof DeliveryNotePropTypes>;

DeliveryIntroduction.prototype = DeliveryNotePropTypes;

export default injectIntl(DeliveryIntroduction);
