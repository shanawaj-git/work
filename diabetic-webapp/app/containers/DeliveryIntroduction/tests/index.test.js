import { fireEvent, render } from 'react-testing-library';

import { IntlProvider } from 'react-intl';
import React from 'react';
import { act } from 'react-test-renderer';
import DeliveryIntroduction from '../index';
jest.mock('react-router-dom', () => ({
  __esModule: true,
  useLocation: jest.fn().mockReturnValue({
    pathname: '/review-prescription',
    search: '',
    hash: '',
    state: null,
    key: '',
  }),
}));

const WrappedDeliveryIntroduction = () => (
  <IntlProvider locale="en">
    <DeliveryIntroduction />
  </IntlProvider>
);
const NAVIGATION_ARGS = { pathname: 'address-selection' };
describe('<WrappedDeliveryIntroduction />', () => {
  beforeAll(() => {
    jest.useFakeTimers();
  });

  afterAll(() => {
    jest.useRealTimers();
  });

  it('should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(<WrappedDeliveryIntroduction />);
    expect(firstChild).toMatchSnapshot();
  });

  it('should go to next screen when iUnderstand button is clicked', () => {
    const mockHistory = {
      push: jest.fn(),
    };

    const mockLocation = {
      search: '',
    };

    const props = {
      location: mockLocation,
    };

    act(() => {
      const tree = render(
        <IntlProvider locale="en">
          <DeliveryIntroduction history={mockHistory} {...props} />
        </IntlProvider>,
      );
      const iUnderstandButton = tree.getByTestId('iUnderstand-btn');
      fireEvent.click(
        iUnderstandButton,
        new MouseEvent('click', {
          bubbles: true,
          cancelable: true,
        }),
      );
      expect(mockHistory.push).toBeCalledWith(NAVIGATION_ARGS);
    });
  });
  it('should open modal when hyperlink(why must i be present in person?) is clicked', () => {
    const tree = render(<WrappedDeliveryIntroduction />);
    const linkElement = tree.getByTestId('modal-link');

    fireEvent.click(linkElement);
    tree.rerender(<WrappedDeliveryIntroduction />);
    tree.getByTestId('modal');

    const buttonElement = tree.getByTestId('close-button');
    expect(buttonElement).toBeDefined();
    expect(tree.container).toMatchSnapshot();
  });
  it('should close model when back-button is clicked', async () => {
    const tree = render(<WrappedDeliveryIntroduction />);
    const linkElement = tree.getByTestId('modal-link');
    expect(linkElement).toBeTruthy();
    fireEvent.click(linkElement);
    tree.rerender(<WrappedDeliveryIntroduction />);
    const modalElement = tree.getByTestId('modal');
    expect(modalElement).toBeTruthy();

    const backButton = tree.getByTestId('close-button');
    expect(backButton).toBeDefined();
    fireEvent(
      backButton,
      new MouseEvent('click', {
        bubbles: true,
        cancelable: true,
      }),
    );

    tree.rerender(<WrappedDeliveryIntroduction />);
    expect(tree.getByTestId('modal-link')).toBeTruthy();
  });
});
