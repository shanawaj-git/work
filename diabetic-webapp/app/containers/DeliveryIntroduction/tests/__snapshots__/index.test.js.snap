// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<WrappedDeliveryIntroduction /> should open modal when hyperlink(why must i be present in person?) is clicked 1`] = `
<div>
  <div
    class="sc-ifAKCX dPPXDZ"
  >
    <div
      class="sc-bZQynM cQtRSK"
    >
      <h2
        class="MuiTypography-root MuiTypography-h2 sc-jzJRlG dVWWoz css-1jfg05l-MuiTypography-root"
      >
        Be present at
      </h2>
      <h1
        class="MuiTypography-root MuiTypography-h1 sc-cSHVUG kPaRAP css-qu1isr-MuiTypography-root"
      >
        the time of delivery 
      </h1>
    </div>
    <div
      class="sc-EHOje byttaB"
    >
      <span
        class="MuiTypography-root MuiTypography-body1 sc-fjdhpX jPjbBh css-1reifv3-MuiTypography-root"
      >
        <p>
          You are required to 
          <strong>
            present in person 
          </strong>
          at the time of delivery.Please select a date and the time accordingly.
        </p>
      </span>
    </div>
    <div
      class="sc-jTzLTM eGUQRX"
    />
    <div
      class="sc-dnqmqq bLaDla"
    >
      <div
        class="sc-iwsKbI fZUtNN"
      >
        <img
          alt="header_img"
          src="IMAGE_MOCK"
        />
      </div>
      <div
        class="sc-gZMcBi kopsXO"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 sc-gqjmRU eRtcfe css-1reifv3-MuiTypography-root"
        >
          Step 1
        </p>
        <p
          class="MuiTypography-root MuiTypography-body1 sc-VigVT ccGNxA css-1reifv3-MuiTypography-root"
        >
          Select a date,time and place for delivery,so you can receive your meds in person.
        </p>
      </div>
    </div>
    <div
      class="sc-dnqmqq bLaDla"
    >
      <div
        class="sc-iwsKbI fZUtNN"
      >
        <img
          alt="header_img"
          src="IMAGE_MOCK"
        />
      </div>
      <div
        class="sc-gZMcBi kopsXO"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 sc-gqjmRU eRtcfe css-1reifv3-MuiTypography-root"
        >
          Step 2
        </p>
        <p
          class="MuiTypography-root MuiTypography-body1 sc-VigVT ccGNxA css-1reifv3-MuiTypography-root"
        >
          Setup your refill reminder so your health doesnt miss out 
        </p>
      </div>
    </div>
    <div
      class="sc-gzVnrw gUmHAA"
    >
      <button
        class="root StyledButton-sc-1apep4b-0 hPQGaD MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root  css-sghohy-MuiButtonBase-root-MuiButton-root"
        data-testid="iUnderstand-btn"
        tabindex="0"
        type="button"
      >
        <div
          style="visibility: visible;"
        >
          I Understand,Proceed
        </div>
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
    <p
      class="MuiTypography-root MuiTypography-body1 sc-htoDjs dnbmtK css-1reifv3-MuiTypography-root"
    >
      <button
        data-testid="modal-link"
        id="contactSupport"
        type="button"
      >
        Why must i be present in person?
      </button>
    </p>
  </div>
</div>
`;

exports[`<WrappedDeliveryIntroduction /> should render and match the snapshot 1`] = `
<div
  class="sc-ifAKCX dPPXDZ"
>
  <div
    class="sc-bZQynM cQtRSK"
  >
    <h2
      class="MuiTypography-root MuiTypography-h2 sc-jzJRlG dVWWoz css-1jfg05l-MuiTypography-root"
    >
      Be present at
    </h2>
    <h1
      class="MuiTypography-root MuiTypography-h1 sc-cSHVUG kPaRAP css-qu1isr-MuiTypography-root"
    >
      the time of delivery 
    </h1>
  </div>
  <div
    class="sc-EHOje byttaB"
  >
    <span
      class="MuiTypography-root MuiTypography-body1 sc-fjdhpX jPjbBh css-1reifv3-MuiTypography-root"
    >
      <p>
        You are required to 
        <strong>
          present in person 
        </strong>
        at the time of delivery.Please select a date and the time accordingly.
      </p>
    </span>
  </div>
  <div
    class="sc-jTzLTM eGUQRX"
  />
  <div
    class="sc-dnqmqq bLaDla"
  >
    <div
      class="sc-iwsKbI fZUtNN"
    >
      <img
        alt="header_img"
        src="IMAGE_MOCK"
      />
    </div>
    <div
      class="sc-gZMcBi kopsXO"
    >
      <p
        class="MuiTypography-root MuiTypography-body1 sc-gqjmRU eRtcfe css-1reifv3-MuiTypography-root"
      >
        Step 1
      </p>
      <p
        class="MuiTypography-root MuiTypography-body1 sc-VigVT ccGNxA css-1reifv3-MuiTypography-root"
      >
        Select a date,time and place for delivery,so you can receive your meds in person.
      </p>
    </div>
  </div>
  <div
    class="sc-dnqmqq bLaDla"
  >
    <div
      class="sc-iwsKbI fZUtNN"
    >
      <img
        alt="header_img"
        src="IMAGE_MOCK"
      />
    </div>
    <div
      class="sc-gZMcBi kopsXO"
    >
      <p
        class="MuiTypography-root MuiTypography-body1 sc-gqjmRU eRtcfe css-1reifv3-MuiTypography-root"
      >
        Step 2
      </p>
      <p
        class="MuiTypography-root MuiTypography-body1 sc-VigVT ccGNxA css-1reifv3-MuiTypography-root"
      >
        Setup your refill reminder so your health doesnt miss out 
      </p>
    </div>
  </div>
  <div
    class="sc-gzVnrw gUmHAA"
  >
    <button
      class="root StyledButton-sc-1apep4b-0 hPQGaD MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root  css-sghohy-MuiButtonBase-root-MuiButton-root"
      data-testid="iUnderstand-btn"
      tabindex="0"
      type="button"
    >
      <div
        style="visibility: visible;"
      >
        I Understand,Proceed
      </div>
    </button>
  </div>
  <p
    class="MuiTypography-root MuiTypography-body1 sc-htoDjs dnbmtK css-1reifv3-MuiTypography-root"
  >
    <button
      data-testid="modal-link"
      id="contactSupport"
      type="button"
    >
      Why must i be present in person?
    </button>
  </p>
</div>
`;
